src/main/java/woowacourse/shoppingcart/application/CustomerService.java

```diff
@@ -23,36 +27,49 @@ public CustomerService(final CustomerDao customerDao, final Encryptor encryptor)
     }
 
     public void addCustomer(final CustomerRequest customerRequest) {
-        if (customerDao.existsByName(customerRequest.getName())) {
+        Customer customer = new Customer(customerRequest.getName(), customerRequest.getPassword(), encryptor);
+        if (customerDao.existsByName(customer.getName())) {
             throw new DuplicateNameException();
         }
-        String encryptedPassword = encryptor.encrypt(customerRequest.getPassword());
-        Customer customer = new Customer(customerRequest.getName(), encryptedPassword);
         customerDao.save(customer);
     }
 
-    public void deleteCustomerByName(final String customerName) {
-        customerDao.deleteByName(customerName);
+    public void deleteCustomerByName(final String rawUserName) {
+        UserName userName = new UserName(rawUserName);
+        if (customerDao.existsByName(userName)) {
+            customerDao.deleteByName(userName);
+            return;
+        }
+        throw new NoExistUserException();
```

Comment:
```suggestion
        if (!customerDao.existsByName(userName)) {
            throw new NoExistUserException();
        }
        customerDao.deleteByName(userName);
```

이렇게 순서를 바꾸는게 코드가 좀 더 간결하게 보여질 것 같아요!! 한번 참고 부탁드려요 :)

Reply:
제 생각으로는 if문에 !를 쓰는 것이 생각보다 잘 안읽힐 거 같아 throw문을 밖으로 빼는 방식으로 구성하였습니다😅